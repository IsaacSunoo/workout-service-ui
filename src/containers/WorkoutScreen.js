import React, { Component } from 'react';
import { Text, View, StyleSheet, ScrollView } from 'react-native';
import { connect } from 'react-redux';

export class WorkoutScreen extends Component {

  render() {
    const { name, length, repetitions } = this.props.workout;

    return (
      <ScrollView>
        <View style={styles.container}>
          <Text>{name}</Text>
          <Text>Length: {length}</Text>
        </View>

        <View style={styles.exercise}>
          { repetitions.map(exercise => {
            return  <Text style={styles.textStyle}>{exercise.description}: {exercise.exercise.name}</Text>
          }) }
        </View>
      </ScrollView>
    )
  }
}

export const mapStateToProps = (state) => ({
  workout: state.selectedWorkout
})

export default connect(mapStateToProps)(WorkoutScreen)

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center'
  },
  exercise: {
    // flex: 1,
  },
  textStyle: {
    textAlign: 'center',
    flex: 2,
    flexDirection: 'row',
  }

});