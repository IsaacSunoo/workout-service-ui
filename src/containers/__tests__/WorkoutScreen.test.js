import React from 'react';
import { shallow } from 'enzyme';
import { WorkoutScreen } from '../WorkoutScreen';

describe('WorkoutScreen', () => {

  let wrapper;
  let navigationMock = { getParam: jest.fn() };
  let mockWorkout = {
    item: {
      id: 1,
      name: "Workout 1",
      length: 30,
      avgrating: 3.5,
    }
  }
  

  beforeEach(() => {
    wrapper = shallow(
      <WorkoutScreen
        navigation={navigationMock}
        workout={mockWorkout}
      />
)
  })

  it.skip('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})