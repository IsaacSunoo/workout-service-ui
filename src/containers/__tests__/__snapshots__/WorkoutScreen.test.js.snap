// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkoutScreen should match snapshot 1`] = `undefined`;
